packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx

```diff
@@ -0,0 +1,359 @@
+import * as React from "react";
+import { get } from "lodash-es";
+import { canUseDOM } from "exenv-es6";
+import { IHorizontalOverflowClassNameContract, IManagedClasses } from "@microsoft/fast-components-class-name-contracts-base";
+import { getClientRectWithMargin } from "@microsoft/fast-web-utilities";
+import Foundation, { HandledProps } from "../foundation";
+import { HorizontalOverflowProps, IHorizontalOverflowHandledProps } from "./horizontal-overflow.props";
+
+export enum ButtonDirection {
+    previous = "previous",
+    next = "next"
+}
+
+export interface IHorizontalOverflowState {
+    itemsHeight: number;
+}
+
+export type AnimateScroll = () => void;
+
+class HorizontalOverflow extends Foundation<HorizontalOverflowProps,  React.AllHTMLAttributes<HTMLDivElement>, IHorizontalOverflowState> {
+
+    protected handledProps: HandledProps<IHorizontalOverflowHandledProps & IManagedClasses<IHorizontalOverflowClassNameContract>> = {
+        children: void 0,
+        scrollDuration: void 0,
+        managedClasses: void 0
+    };
+
+    private horizontalOverflowItemsRef: React.RefObject<HTMLUListElement>;
+
+    constructor(props: HorizontalOverflowProps) {
+        super(props);
+
+        this.horizontalOverflowItemsRef = React.createRef();
+
+        this.state = {
+            itemsHeight: 0
+        };
+    }
+
+    /**
+     * Renders the Horizontal Overflow markup
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div
+                className={this.props.managedClasses.horizontalOverflow}
+                onLoad={this.itemsOnLoad}
+            >
+                <div style={{height: `${this.state.itemsHeight}px`, position: "relative", overflow: "hidden"}}>
+                    <ul
+                        className={this.props.managedClasses.horizontalOverflow_items}
+                        style={{position: "relative", whiteSpace: "nowrap", overflowX: "scroll", padding: 0, margin: 0}}
+                        ref={this.horizontalOverflowItemsRef}
+                    >
+                        {this.getItems()}
+                    </ul>
+                </div>
+                <div className={this.props.managedClasses.horizontalOverflow_previous}>
+                    <div>
+                        <div onClick={this.handlePreviousClick}>
+                            {this.getChildBySlot(ButtonDirection.previous)}
+                        </div>
+                    </div>
+                </div>
```

Comment:
Do we need the additional `<div>` wrapping the div with the onClick? It seems unneccessary without more context.